tests/components/tod/test_binary_sensor.py

```diff
@@ -862,27 +864,257 @@ async def test_sun_offset(hass):
 
 async def test_dst(hass):
     """Test sun event with offset."""
-    hass.config.time_zone = "CET"
-    test_time = datetime(2019, 3, 30, 3, 0, 0, tzinfo=dt_util.UTC)
-    config = {
-        "binary_sensor": [
-            {"platform": "tod", "name": "Day", "after": "2:30", "before": "2:40"}
-        ]
-    }
-    # Test DST:
-    # after 2019-03-30 03:00 CET the next update should ge scheduled
-    # at 3:30 not 2:30 local time
-    entity_id = "binary_sensor.day"
-    with patch(
-        "homeassistant.components.tod.binary_sensor.dt_util.utcnow",
-        return_value=test_time,
-    ):
-        await async_setup_component(hass, "binary_sensor", config)
-        await hass.async_block_till_done()
+    try:
+        hass.config.time_zone = "CET"
+        dt_util.set_default_time_zone(dt_util.get_time_zone("CET"))
+        test_time = datetime(2019, 3, 30, 3, 0, 0, tzinfo=dt_util.UTC)
+        config = {
+            "binary_sensor": [
+                {"platform": "tod", "name": "Day", "after": "2:30", "before": "2:40"}
+            ]
+        }
+        # Test DST:
+        # after 2019-03-30 03:00 CET the next update should ge scheduled
+        # at 3:30 not 2:30 local time
+        entity_id = "binary_sensor.day"
+        with patch(
+            "homeassistant.components.tod.binary_sensor.dt_util.utcnow",
+            return_value=test_time,
+        ):
+            await async_setup_component(hass, "binary_sensor", config)
+            await hass.async_block_till_done()
+
+            await hass.async_block_till_done()
+            state = hass.states.get(entity_id)
+            assert state.attributes["after"] == "2019-03-31T03:30:00+02:00"
+            assert state.attributes["before"] == "2019-03-31T03:40:00+02:00"
+            assert state.attributes["next_update"] == "2019-03-31T03:30:00+02:00"
+            assert state.state == STATE_OFF
+    finally:
+        dt_util.set_default_time_zone(ORIG_TIMEZONE)
```

Comment:
Can we make a pytest fixture that has `autouse=True` and yields and then resets the original timezone? Then we don't need to indent the whole test code here.